### DISCUSSION TITLE
Guitar: capo, tuning and transposed instrument

### DISCUSSION THREAD

**Discussion by ShibuyaCyana** [musescore/MuseScore#20476] [Ideas]
We could have these tickboxes for guitar treble clef properties:
- transposed notation with capo
    - transpose key signature
- transposed notation with non-standard tuning
    - transpose key signature
 
and this for tablature:
- fret number count from capo

This is the current state with 4.2:
1. Guitar with capo is treated as transposed instrument, not responding to concert pitch and causing [this issue](https://github.com/musescore/MuseScore/issues/19860);
2. Alternative tuned guitar is NOT treated as transposed instrument. Manually set transpose will mess up tablature numbers (looks like they're directly calculated from notated pitch and string tuning, ignoring transposition);
3. With capo, tablature numbers are also "transposed", which is following guitar pro behavior.

Transposed notation of both treble clef and tab won't make much sense if it's a partial capo; and, personally, I prefer "absolute" tab numbers with capo, since I find frets with side dot or muscle memory, and none of them will move with capo.
(I don't use capo often, I guess this kind of notation makes sense if mostly playing close to capo)

So:
1. Support concert pitch notation with capo, should also help solving the aforementioned issue.
1.1. transposed notation may optionally use transposed key signature
2. Support transposed notation for down tuned guitar, which would also benefit things like baritone guitar, as long as following standard tuning interval;
3. Support optional absolute fret number with capo, which should also make partial capo easier to read.

Since notation for guitar has no globally agreed rules, I think it would be better to put these option in staff properties, rather than responding to the global concert pitch toggle.

**Comment by scorster** [comment]
Re: **1(a)  Guitar with capo is treated as transposed instrument, not [corresponding to written] concert pitch**

Well said.

Re: **3(b)  Support optional absolute fret number with capo**

There's [an in depth discussion about absolute fret numbering on the .org forum and specifically how it pertains to a partial capo.](https://musescore.org/en/node/353203#comment-1207748) Thought I'd mention in case you hadn't seen it.

I agree on the importance of absolute fret numbering—as an option. Setting it as a staff property makes sense when a score only requires absolute tablature fret numbering. But it's worth noting: Prior to MuseScore 4.2 tablature numbering was always expressed relative to the capo.

However ( as discussed on the forum) with partial capo notation, situations arise where **we need a mix of capo-relative and absolute fret numbering.** In such cases we a need a "fret numbering" property per string (i.e. capo relative numbering vs. absolute fret numbering) that takes priority over the staff/part setting. 

For instance, most scorists likely expect frets in partial capo tablature expressed relative to the capo. However, when there's a section of the part to be played "below" the partial capo, absolute fret numbers provide a way of specifying (without resorting to negative frets relative to the capo.) 

This is discussed in the cited forum post and supported with illustrations and suggested nomenclature. While the concept of absolute frets is straight forward, mixing absolute with capo-relative is uncharted territory (i.e. there are no globally agreed upon rules). My spidey sensibilities say, when there's a need to fret behind the capo, absolute fret numbers are more intuitive than negative numbers relative to the capo. But ... options are always good. 

As stated in the original post, there is no standard for partial capo, but MuseScore could forge an intelligent path forward to address the full range of partial capo notation needs, including mixes of capo-relative fret numbers and the occasional absolute fret numbering (i.e. nut relative.) And the user would be in the driver's seat if the option is in:


     • The capo popup (my vote!)

     —or —

     • Staff/Part Properties rather than a Global toggle. Ideally it would be in Edit String tuning.

scorster

**Comment by scorster** [comment]
MS4.2 bestowed upon us the absolutely wonderful partial capo functionality, but it lacks the option for "capo relative fret numbering". Thus it remains incomplete and generally unusable for me ... in kind, this "absolute" notational approach is unusable for anyone who needs uncapoed strings notated "relative to the capo" which is likely the preferred approach.  

The issues have been well outlined in various discussions, but I'd don't see the **capo relative vs. nut relative (absolute) fret numbering** assigned to any Github MuseScore project. Am I missing something?

[This related issue](https://github.com/musescore/MuseScore/issues/19860) is marked P1 since Oct 2023. But says needs review.

Ideally MuseScore will offer a setting to toggle between relative vs. absolute. This could be per score, per staff, per selection, and/or a per note.

scorster

**Comment by scorster** [comment]
It's wonderful to see MuseScore's recent advancements in some areas of guitar notation, like fingerings in Fretboard diagrams ... and the Fretboard Diagram Legend in MS 3.6.3.

However, best I can tell, there's never been Github request regarding the points in this discussion. So I'm planning to start one.
